x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -91,6 +94,79 @@ func (k Keeper) DelegateOnHost(ctx sdk.Context, hostZone types.HostZone, amt sdk
 	return nil
 }
 
+func (k Keeper) ReinvestRewards(ctx sdk.Context, hostZone types.HostZone) error {
+	_ = ctx
+	// the relevant ICA is the delegate account
+	owner := types.FormatICAAccountOwner(hostZone.ChainId, types.ICAAccountType_DELEGATION)
+	portID, err := icatypes.NewControllerPortID(owner)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "%s has no associated portId", owner)
+	}
+	connectionId, err := k.GetConnectionId(ctx, portID)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidChainID, "%s has no associated connection", portID)
+	}
+
+	// Fetch the relevant ICA
+	delegationAccount := hostZone.GetDelegationAccount()
+	withdrawAccount := hostZone.GetWithdrawalAccount()
+
+	SubmitTxs := k.SubmitTxs
+	GetParam := k.GetParam
+	cdc := k.cdc
+
+	var cb icqkeeper.Callback = func(k icqkeeper.Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
```

Comment:
k => icqk so that we don't override the parent keeper definition and can distinguish

Reply:
Done.